pallets/multicollateral-bonding-curve-pool/src/lib.rs

```diff
@@ -1594,29 +1594,29 @@ impl<T: Config> LiquiditySource<T::DEXId, T::AccountId, T::AssetId, Balance, Dis
         input_asset_id: &T::AssetId,
         output_asset_id: &T::AssetId,
         amount: QuoteAmount<Balance>,
-        samples_count: usize,
+        recommended_samples_count: usize,
         deduce_fee: bool,
-    ) -> Result<VecDeque<SwapChunk<Balance>>, DispatchError> {
+    ) -> Result<(VecDeque<SwapChunk<Balance>>, Weight), DispatchError> {
         if !Self::can_exchange(dex_id, input_asset_id, output_asset_id) {
             fail!(Error::<T>::CantExchange);
         }
         if amount.amount().is_zero() {
-            return Ok(VecDeque::new());
+            return Ok((VecDeque::new(), Weight::zero()));
```

Comment:
zero is okay

Reply:
Ok, I guess it shouldn't be a big problem, but my point was that calculations done before this return should be considered in weight.

It just seems for me as an easy approach for weight evaluation (particularly, since it's quite [important](https://docs.substrate.io/test/benchmark/#why-benchmark-a-pallet) to do so). I think it's easier to consider all computations that happen in final weight rather than checking case-by-case whether zero weight here can lead to some attacks or not (although in this the answer is "likely not", so I'm going to approve)